apps/rush-lib/src/cli/actions/PublishAction.ts

```diff
@@ -453,4 +472,28 @@ export class PublishAction extends BaseRushAction {
 
     return `${name}-${project.packageJson.version}.tgz`;
   }
-}
\ No newline at end of file
+
+  private _manageLooseVersions(action: 'lock' | 'unlock'): void {
+    const projectsByName: Map<string, RushConfigurationProject> = this.rushConfiguration.projectsByName;
+
+    for (const [, project] of projectsByName.entries()) {
+      const editor: PackageJsonEditor = project.packageJsonEditor;
+
+      const dependencies: ReadonlyArray<PackageJsonDependency> = editor.dependencyList;
+
+      for (const dependency of dependencies) {
+        const localProject: RushConfigurationProject | undefined = projectsByName.get(dependency.name);
+
+        if (localProject) {
+          if (action === 'lock') {
+            dependency.setVersion(localProject.packageJson.version);
+          } else {
+            dependency.setVersion('*');
```

Comment:
I think we should remember whatever specifier was there before and put that back when versions get unlocked instead of assuming the repo is using `*` specifiers.

Reply:
In the case of my team's repo, I actually want this logic to force all the dependencies to `*`, basically ensuring they're always stored in this fashion. I think it's actually a feature that this would act as a 'linter' on inter-project dependencies.